src/OracleService/OracleService.cs

```diff
@@ -314,18 +327,23 @@ private async Task<(OracleResponseCode, string)> ProcessUrlAsync(string url)
             }
         }
 
-        public static Transaction CreateResponseTx(DataCache snapshot, OracleRequest request, OracleResponse response, ECPoint[] oracleNodes, ProtocolSettings settings)
+        public static Transaction CreateResponseTx(DataCache snapshot, OracleRequest request, OracleResponse response, ECPoint[] oracleNodes, ProtocolSettings settings, bool useCurrentHeight = false)
         {
             var requestTx = NativeContract.Ledger.GetTransactionState(snapshot, request.OriginalTxid);
             var n = oracleNodes.Length;
             var m = n - (n - 1) / 3;
             var oracleSignContract = Contract.CreateMultiSigContract(m, oracleNodes);
-
+            uint height = NativeContract.Ledger.CurrentIndex(snapshot);
```

Comment:
I think that this can produce a different responseTx, if the tx doesn't arrive in the same block to all nodes, it coud produce an error. We should try to unify this

Reply:
The probability of produceing diffrent responseTx is about 1/5760. When it happens, it will be removed from queue when reach "MaxTaskTimeout", then the request will re-enqueue the ProcessRequestAsync method. 
These steps can repeat many times until this oracle request was processed successfully.

Reply:
I think that we should reduce this probability, during this time we have the transaction in memory all because this field.

Reply:
Yes. There are still some more serious problems with current oracle architecture. See https://github.com/neo-project/neo-modules/issues/611 .
I think we should refact the oracle service to resolve these problems.

Reply:
I think this particular problem is actually way less probable than 1/5760. It's only when `height > validUntilBlock` the increment is triggered and for this to happen we have to be already `MaxValidUntilBlockIncrement` blocks away from `requestTx.BlockIndex` which is a 100 blocks. On a healthy network I'd expect most of oracle requests to be handled within 1-2 blocks from `requestTx.BlockIndex` with maybe something like 99% of requests completing in 10 blocks. Then of course there is a long tail of less successful requests with some crossing this `MaxValidUntilBlockIncrement` line, but even then oracle nodes would have to be a little unsynchronized to trigger different response tx problem. Then when every oracle nodes crosses `requestTx.BlockIndex + MaxValidUntilBlockIncrement` there is another ~`MaxValidUntilBlockIncrement` window of time where this problem won't occur.

So it should be fine as a fallback mechanism (which it is now). Not perfect, but this problem doesn't seem to have any perfect solution.

#611 is more interesting though.

Reply:
If two nodes produces a different `ValidUntilBlock`, his signatures wont be valid. So if the tx arrives in different blocks to oracles nodes (I think that it could be very easier) it won't be valid.

Reply:
> if the tx arrives in different blocks to oracles nodes

But isn't `requestTx.BlockIndex` is a block number that contained `requestTx` and there could only be one block with this tx ever?
```
            var requestTx = NativeContract.Ledger.GetTransactionState(snapshot, request.OriginalTxid);
```

Reply:
You are right, I saw a different thing 😆